pkg/workflows/workflow.go

```diff
@@ -392,15 +392,14 @@ func (w *workflowLifecycle) processWorkflowResources(workflowStepObject interfac
 		if foundManifests {
 			var objs []string
 			for _, obj := range strings.Split(manifests.(string), "---\n") {
-				resource := &unstructured.Unstructured{}
-				data, err := w.processArtifact(obj, resource, wt)
+				data, err := w.processArtifact(obj, &unstructured.Unstructured{}, wt)
 				if err != nil {
 					return err
 				}
 				objs = append(objs, data)
 			}
-			manifests = strings.Join(objs, "---\n")
-			err = unstructured.SetNestedField(workflowStepObject.(map[string]interface{}), manifests.(string), "resource", "manifest")
+			manifests = strings.Join(objs, "\n")
```

Comment:
Is this change purely for b/c of the bug w/ manifests seperated by `---`?

Reply:
it is adjusted to get through argo-workflow resource validation